packages/ingest/src/steps/1-captions.ts

```diff
@@ -0,0 +1,140 @@
+import { PrismaClientWithAccelerate } from "@oakai/db";
+
+import { IngestError } from "../IngestError";
+import { getCaptionsByFileName } from "../captions/getCaptionsByFileName";
+import { getCaptionsFileNameForLesson } from "../captions/getCaptionsFileNameForLesson";
+import { createCaptionsRecord } from "../db-helpers/createCaptionsRecord";
+import { createErrorRecord } from "../db-helpers/createErrorRecord";
+import { getIngestById } from "../db-helpers/getIngestById";
+import { loadLessonsAndUpdateState } from "../db-helpers/loadLessonsAndUpdateState";
+import { Step, getPrevStep } from "../db-helpers/step";
+import { updateLessonsState } from "../db-helpers/updateLessonsState";
+import { Captions } from "../zod-schema/zodSchema";
+
+const currentStep: Step = "captions_fetch";
+const prevStep = getPrevStep(currentStep);
+
+/**
+ * This function fetches and stores captions for the lessons imported in the most recent ingest.
+ */
+export async function captions({
+  prisma,
+  ingestId,
+}: {
+  prisma: PrismaClientWithAccelerate;
+  ingestId: string;
+}) {
+  const ingest = await getIngestById({ prisma, ingestId });
+
+  const lessons = await loadLessonsAndUpdateState({
+    prisma,
+    ingestId,
+    prevStep,
+    currentStep,
+  });
+
+  if (ingest.config.sourcePartsToInclude === "title-subject-key-stage") {
+    console.log("Skipping captions fetch for title-subject-key-stage ingest");
```

Comment:
It would be nice to have a follow up task to see if it's a good idea to move these logs to the new logger. It might not make sense if this package is only run in isolation of our other packages

Reply:
💯